HLTrigger/Egamma/plugins/HLTScoutingEgammaProducer.cc

```diff
@@ -16,6 +16,29 @@ Description: Producer for ScoutingElectron and ScoutingPhoton
 
 #include "HLTScoutingEgammaProducer.h"
 
+// function to find rechhit associated to detid and return energy
+float recHitE(const DetId id, const EcalRecHitCollection& recHits) {
+  if (id == DetId(0)) {
+    return 0;
+  } else {
+    EcalRecHitCollection::const_iterator it = recHits.find(id);
+    if (it != recHits.end())
+      return (*it).energy();
+  }
+  return 0;
```

Comment:
Isn't the `if (id == DetId(0))` check unnecessary ?
If the `recHits` map don't have any associated to the `0` detId, the function would return `0` anyway.

No changes needed for this PR, but it could be cleaned up in a later PR.